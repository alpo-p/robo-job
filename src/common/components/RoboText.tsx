import styled from 'styled-components/native'
import sharedStyles from '../styles'

type FontSize = 'tiny' | 'small' | 'medium' | 'large'
type Colors = 'primaryWhite' | 'primaryDark' | 'primaryColorful'
type Weight = 'light' | 'normal' | 'bold'
interface Props {
  size?: FontSize
  color?: Colors
  weight?: Weight
  uppercase?: boolean
}

const RoboText = styled.Text<Props>`
  font-family: ${({ weight }) => {
    if (weight === 'light') return 'DamascusLight'
    if (weight === 'bold') return 'DamascusBold'
    return 'Damascus'
  }};
  font-size: ${({ size }) => {
    if (size === 'tiny') return '12px'
    if (size === 'small') return '14px'
    if (size === 'medium') return '16px'
    if (size === 'large') return '20px'
    return '14px'
  }};
  color: ${({ color }) => {
    if (color === 'primaryColorful') return sharedStyles.primaryColor
    if (color === 'primaryDark') return sharedStyles.darkGrey
    return sharedStyles.white
  }};
  text-transform: ${({ uppercase }) => {
    if (uppercase) return 'uppercase'
    return 'none'
  }};
`

export default RoboText
