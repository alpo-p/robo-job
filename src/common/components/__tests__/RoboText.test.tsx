import React from 'react'
import renderer from 'react-test-renderer'
import RoboText from '../RoboText'

describe('RoboText', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<RoboText>Testi</RoboText>)
    expect(tree).toMatchSnapshot()
  })
})
