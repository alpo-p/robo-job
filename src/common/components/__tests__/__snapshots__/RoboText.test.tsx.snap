// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoboText should match snapshot 1`] = `
<Text
  style={
    Object {
      "color": "rgb(28, 28, 30)",
      "fontFamily": "Damascus",
      "fontSize": 14,
      "textTransform": "none",
    }
  }
>
  Testi
</Text>
`;
